rootex/framework/components/physics/box_component.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "physics_component.h"
+
+#include "btBulletDynamicsCommon.h"
+
+class BoxComponent : public PhysicsComponent
+{
+	static Component* Create(const JSON::json& boxComponentData);
+	static Component* CreateDefault();
+
+	Vector3 m_Dimensions;
+
+	friend class EntityFactory;
+
+public:
+	BoxComponent(Vector3 dimensions, const String& matName);
+
+	Vector3 getDimensions() const { return m_Dimensions; }
+
+	virtual String getName() const override { return "BoxComponent"; };
+	virtual JSON::json getJSON() const override;
```

Comment:
Add a draw() function as in all the other components. We want the ability to change the data inside this component from the editor. The overridable `Component::draw()` is meant for that